src/app/shared/modals/change-password/change-password.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, AbstractControl, Validators, ValidationErrors } from '@angular/forms';
+
+@Component({
+  selector: 'app-change-password',
+  templateUrl: './change-password.component.html',
+  styleUrls: ['./change-password.component.scss']
+})
+export class ChangePasswordComponent implements OnInit {
+  changeForm: FormGroup;
+  private readonly pswrd: RegExp = /^(?=\D*\d)(?=[^a-z]*[a-z])(?=[^A-Z]*[A-Z]).{8,30}$/;
+
+  constructor(private readonly fb: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.changeForm = this.fb.group({
+      oldPassword: this.fb.control('', Validators.required),
+      newPassword: this.fb.control('', [Validators.required, this.psswrdValidator.bind(this)]),
```

Comment:
`psswrdValidator` - please rename to `passwordValidator`, use the full word in naming in feature

`.bind(this)` - using `bind` is bad practice, I recommend moving this validator to a separate file as an export function

```
export function passwordValidator(control: AbstractControl): ValidationErrors | null {
  if (control?.value) {
    const isValid = PASSWORD_REGEX.test(control.value);
    return !isValid ? { invalidPassword: true } : null;

  }

  return null;
}
```
and user as 
```
...
password: this.fb.control('', [Validators.required, passwordValidator]),
...
```

Reply:
also move this regex `private readonly pswrd: RegExp = /^(?=\D*\d)(?=[^a-z]*[a-z])(?=[^A-Z]*[A-Z]).{8,30}$/;` to file with validator

Reply:
fixed